### ISSUE TITLE
auto_lightgbm not working

### ISSUE THREAD

**Issue by jarichb** [functime-org/functime#190]
**Describe the bug**
auto_lightgbm is not working, when following the documentation: https://docs.functime.ai/forecasting/#quantile-regression

**To Reproduce**

```python
import json
from timeit import default_timer

import polars as pl

from functime.cross_validation import train_test_split
from functime.forecasting import auto_linear_model, linear_model, naive, snaive, auto_lightgbm
from functime.metrics import smape
from functime.preprocessing import scale
from functime.seasonality import add_fourier_terms

start_time = default_timer()

y = pl.read_parquet(
    "https://github.com/functime-org/functime/raw/main/data/commodities.parquet"
)
entity_col, time_col = y.columns[:2]
X = y.select([entity_col, time_col]).pipe(add_fourier_terms(sp=12, K=6)).collect()

print("🎯 Target variable (y):\n", y)
print("📉 Exogenous variables (X):\n", X)

test_size = 3
freq = "1mo"
y_train, y_test = train_test_split(test_size)(y)
X_train, X_test = train_test_split(test_size)(X)

y_pred_10 = auto_lightgbm(alpha=0.1, freq="1d")(y=y_train, fh=28)
y_pred_90 = auto_lightgbm(alpha=0.9, freq="1d")(y=y_train, fh=28)
```

**Expected behavior**
that the example in the documentation would at least execute

**Screenshots**
<img width="770" alt="Screenshot 2024-04-08 at 21 10 10" src="https://github.com/functime-org/functime/assets/66638094/abcfe4cf-5806-4519-826a-83145ed5199b">
<img width="721" alt="Screenshot 2024-04-08 at 21 09 20" src="https://github.com/functime-org/functime/assets/66638094/cba1b4a3-556a-4f36-9621-660c95aa88f2">

**Desktop (please complete the following information):**
 - OS: Mac
 - Functime version: 0.9.5. 

**Additional context**
As per requested on [LinkedIn](https://www.linkedin.com/feed/update/urn:li:activity:7166739667860537344?commentUrn=urn%3Ali%3Acomment%3A%28activity%3A7166739667860537344%2C7183180272505430017%29&dashCommentUrn=urn%3Ali%3Afsd_comment%3A%287183180272505430017%2Curn%3Ali%3Aactivity%3A7166739667860537344%29)

**Comment by baggiponte** [comment]
Ciao @jarichb, thanks for reporting and sorry for the late reply. I see the error! Did you make sure to install `functime[lgb]`?

We documented that on [this](https://docs.functime.ai/installation/) page: how can we improve that?

I think I need to make this error be raised sooner.

**Comment by jarichb** [comment]
interestingly enough, I can't seem to get it installed like that either.
`(.env) jarich@Jarichs-Air functime % pip install "functime[llm,lgb]"

Requirement already satisfied: functime[lgb,llm] in ./.env/lib/python3.8/site-packages (0.9.5)
Requirement already satisfied: cloudpickle in ./.env/lib/python3.8/site-packages (from functime[lgb,llm]) (3.0.0)
Requirement already satisfied: flaml<3,>=2.0.2 in ./.env/lib/python3.8/site-packages (from functime[lgb,llm]) (2.1.2)
Requirement already satisfied: holidays in ./.env/lib/python3.8/site-packages (from functime[lgb,llm]) (0.46)
Requirement already satisfied: numpy in ./.env/lib/python3.8/site-packages (from functime[lgb,llm]) (1.24.4)
Requirement already satisfied: polars>=0.20.7 in ./.env/lib/python3.8/site-packages (from functime[lgb,llm]) (0.20.21)
Requirement already satisfied: scikit-learn<2,>=1.2.2 in ./.env/lib/python3.8/site-packages (from functime[lgb,llm]) (1.3.2)
Requirement already satisfied: scipy in ./.env/lib/python3.8/site-packages (from functime[lgb,llm]) (1.10.1)
Requirement already satisfied: tqdm in ./.env/lib/python3.8/site-packages (from functime[lgb,llm]) (4.66.2)
Requirement already satisfied: typing-extensions in ./.env/lib/python3.8/site-packages (from functime[lgb,llm]) (4.11.0)
Collecting lightgbm (from functime[lgb,llm])
  Using cached lightgbm-4.3.0.tar.gz (1.7 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Collecting openai (from functime[lgb,llm])
  Using cached openai-1.20.0-py3-none-any.whl.metadata (21 kB)
Collecting tabulate (from functime[lgb,llm])
  Using cached tabulate-0.9.0-py3-none-any.whl.metadata (34 kB)
Collecting tenacity (from functime[lgb,llm])
  Using cached tenacity-8.2.3-py3-none-any.whl.metadata (1.0 kB)
Collecting tiktoken (from functime[lgb,llm])
  Using cached tiktoken-0.6.0-cp38-cp38-macosx_11_0_arm64.whl.metadata (6.6 kB)
Requirement already satisfied: joblib>=1.1.1 in ./.env/lib/python3.8/site-packages (from scikit-learn<2,>=1.2.2->functime[lgb,llm]) (1.4.0)
Requirement already satisfied: threadpoolctl>=2.0.0 in ./.env/lib/python3.8/site-packages (from scikit-learn<2,>=1.2.2->functime[lgb,llm]) (3.4.0)
Requirement already satisfied: python-dateutil in ./.env/lib/python3.8/site-packages (from holidays->functime[lgb,llm]) (2.9.0.post0)
Collecting anyio<5,>=3.5.0 (from openai->functime[lgb,llm])
  Using cached anyio-4.3.0-py3-none-any.whl.metadata (4.6 kB)
Collecting distro<2,>=1.7.0 (from openai->functime[lgb,llm])
  Using cached distro-1.9.0-py3-none-any.whl.metadata (6.8 kB)
Collecting httpx<1,>=0.23.0 (from openai->functime[lgb,llm])
  Using cached httpx-0.27.0-py3-none-any.whl.metadata (7.2 kB)
Collecting pydantic<3,>=1.9.0 (from openai->functime[lgb,llm])
  Using cached pydantic-2.7.0-py3-none-any.whl.metadata (103 kB)
Collecting sniffio (from openai->functime[lgb,llm])
  Using cached sniffio-1.3.1-py3-none-any.whl.metadata (3.9 kB)
Collecting regex>=2022.1.18 (from tiktoken->functime[lgb,llm])
  Using cached regex-2024.4.16-cp38-cp38-macosx_11_0_arm64.whl.metadata (40 kB)
Collecting requests>=2.26.0 (from tiktoken->functime[lgb,llm])
  Using cached requests-2.31.0-py3-none-any.whl.metadata (4.6 kB)
Collecting idna>=2.8 (from anyio<5,>=3.5.0->openai->functime[lgb,llm])
  Using cached idna-3.7-py3-none-any.whl.metadata (9.9 kB)
Collecting exceptiongroup>=1.0.2 (from anyio<5,>=3.5.0->openai->functime[lgb,llm])
  Using cached exceptiongroup-1.2.0-py3-none-any.whl.metadata (6.6 kB)
Collecting certifi (from httpx<1,>=0.23.0->openai->functime[lgb,llm])
  Using cached certifi-2024.2.2-py3-none-any.whl.metadata (2.2 kB)
Collecting httpcore==1.* (from httpx<1,>=0.23.0->openai->functime[lgb,llm])
  Using cached httpcore-1.0.5-py3-none-any.whl.metadata (20 kB)
Collecting h11<0.15,>=0.13 (from httpcore==1.*->httpx<1,>=0.23.0->openai->functime[lgb,llm])
  Using cached h11-0.14.0-py3-none-any.whl.metadata (8.2 kB)
Collecting annotated-types>=0.4.0 (from pydantic<3,>=1.9.0->openai->functime[lgb,llm])
  Using cached annotated_types-0.6.0-py3-none-any.whl.metadata (12 kB)
Collecting pydantic-core==2.18.1 (from pydantic<3,>=1.9.0->openai->functime[lgb,llm])
  Using cached pydantic_core-2.18.1-cp38-cp38-macosx_11_0_arm64.whl.metadata (6.5 kB)
Collecting charset-normalizer<4,>=2 (from requests>=2.26.0->tiktoken->functime[lgb,llm])
  Using cached charset_normalizer-3.3.2-cp38-cp38-macosx_11_0_arm64.whl.metadata (33 kB)
Collecting urllib3<3,>=1.21.1 (from requests>=2.26.0->tiktoken->functime[lgb,llm])
  Using cached urllib3-2.2.1-py3-none-any.whl.metadata (6.4 kB)
Requirement already satisfied: six>=1.5 in ./.env/lib/python3.8/site-packages (from python-dateutil->holidays->functime[lgb,llm]) (1.16.0)
Using cached openai-1.20.0-py3-none-any.whl (292 kB)
Using cached tabulate-0.9.0-py3-none-any.whl (35 kB)
Using cached tenacity-8.2.3-py3-none-any.whl (24 kB)
Using cached tiktoken-0.6.0-cp38-cp38-macosx_11_0_arm64.whl (922 kB)
Using cached anyio-4.3.0-py3-none-any.whl (85 kB)
Using cached distro-1.9.0-py3-none-any.whl (20 kB)
Using cached httpx-0.27.0-py3-none-any.whl (75 kB)
Using cached httpcore-1.0.5-py3-none-any.whl (77 kB)
Using cached pydantic-2.7.0-py3-none-any.whl (407 kB)
Using cached pydantic_core-2.18.1-cp38-cp38-macosx_11_0_arm64.whl (1.7 MB)
Using cached regex-2024.4.16-cp38-cp38-macosx_11_0_arm64.whl (291 kB)
Using cached requests-2.31.0-py3-none-any.whl (62 kB)
Using cached sniffio-1.3.1-py3-none-any.whl (10 kB)
Using cached annotated_types-0.6.0-py3-none-any.whl (12 kB)
Using cached certifi-2024.2.2-py3-none-any.whl (163 kB)
Using cached charset_normalizer-3.3.2-cp38-cp38-macosx_11_0_arm64.whl (119 kB)
Using cached exceptiongroup-1.2.0-py3-none-any.whl (16 kB)
Using cached idna-3.7-py3-none-any.whl (66 kB)
Using cached urllib3-2.2.1-py3-none-any.whl (121 kB)
Using cached h11-0.14.0-py3-none-any.whl (58 kB)
Building wheels for collected packages: lightgbm
  Building wheel for lightgbm (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for lightgbm (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [40 lines of output]
      2024-04-16 21:15:14,301 - scikit_build_core - INFO - RUN: /private/var/folders/fl/qtsnzd1x0k3dn109zhspr_yc0000gn/T/pip-build-env-rnuvs7bi/normal/lib/python3.8/site-packages/cmake/data/bin/cmake --version
      2024-04-16 21:15:14,307 - scikit_build_core - INFO - CMake version: 3.29.2
      *** scikit-build-core 0.8.2 using CMake 3.29.2 (wheel)
      2024-04-16 21:15:14,312 - scikit_build_core - INFO - Build directory: /private/var/folders/fl/qtsnzd1x0k3dn109zhspr_yc0000gn/T/tmpl1sla9l5/build
      *** Configuring CMake...
      2024-04-16 21:15:14,322 - scikit_build_core - INFO - RUN: /private/var/folders/fl/qtsnzd1x0k3dn109zhspr_yc0000gn/T/pip-build-env-rnuvs7bi/normal/lib/python3.8/site-packages/ninja/data/bin/ninja --version
      2024-04-16 21:15:14,420 - scikit_build_core - INFO - Ninja version: 1.11.1
      2024-04-16 21:15:14,422 - scikit_build_core - INFO - RUN: /private/var/folders/fl/qtsnzd1x0k3dn109zhspr_yc0000gn/T/pip-build-env-rnuvs7bi/normal/lib/python3.8/site-packages/cmake/data/bin/cmake -S. -B/var/folders/fl/qtsnzd1x0k3dn109zhspr_yc0000gn/T/tmpl1sla9l5/build -DCMAKE_BUILD_TYPE:STRING=Release -C/var/folders/fl/qtsnzd1x0k3dn109zhspr_yc0000gn/T/tmpl1sla9l5/build/CMakeInit.txt -DCMAKE_MAKE_PROGRAM=/private/var/folders/fl/qtsnzd1x0k3dn109zhspr_yc0000gn/T/pip-build-env-rnuvs7bi/normal/lib/python3.8/site-packages/ninja/data/bin/ninja -D__BUILD_FOR_PYTHON:BOOL=ON
      loading initial cache file /var/folders/fl/qtsnzd1x0k3dn109zhspr_yc0000gn/T/tmpl1sla9l5/build/CMakeInit.txt
      -- The C compiler identification is AppleClang 12.0.5.12050022
      -- The CXX compiler identification is AppleClang 12.0.5.12050022
      -- Detecting C compiler ABI info
      -- Detecting C compiler ABI info - done
      -- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc - skipped
      -- Detecting C compile features
      -- Detecting C compile features - done
      -- Detecting CXX compiler ABI info
      -- Detecting CXX compiler ABI info - done
      -- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
      -- Detecting CXX compile features
      -- Detecting CXX compile features - done
      -- Could NOT find OpenMP_C (missing: OpenMP_C_FLAGS OpenMP_C_LIB_NAMES)
      -- Could NOT find OpenMP_CXX (missing: OpenMP_CXX_FLAGS OpenMP_CXX_LIB_NAMES)
      -- Could NOT find OpenMP (missing: OpenMP_C_FOUND OpenMP_CXX_FOUND)
      -- Found OpenMP_C: -Xpreprocessor -fopenmp -I/opt/homebrew/opt/libomp/include
      -- Found OpenMP_CXX: -Xpreprocessor -fopenmp -I/opt/homebrew/opt/libomp/include
      -- Found OpenMP: TRUE
      -- Performing Test MM_PREFETCH
      -- Performing Test MM_PREFETCH - Failed
      -- Performing Test MM_MALLOC
      -- Performing Test MM_MALLOC - Success
      -- Using _mm_malloc
      -- Configuring done (0.7s)
      -- Generating done (0.0s)
      -- Build files have been written to: /var/folders/fl/qtsnzd1x0k3dn109zhspr_yc0000gn/T/tmpl1sla9l5/build
      *** Building project with Ninja...
      2024-04-16 21:15:15,127 - scikit_build_core - INFO - RUN: /private/var/folders/fl/qtsnzd1x0k3dn109zhspr_yc0000gn/T/pip-build-env-rnuvs7bi/normal/lib/python3.8/site-packages/cmake/data/bin/cmake --build /var/folders/fl/qtsnzd1x0k3dn109zhspr_yc0000gn/T/tmpl1sla9l5/build
      ninja: error: '/opt/homebrew/opt/libomp/lib/libomp.dylib', needed by '/private/var/folders/fl/qtsnzd1x0k3dn109zhspr_yc0000gn/T/pip-install-ugmjnjqi/lightgbm_3ae15dce17b94645a154c1a1980eb331/lib_lightgbm.so', missing and no known rule to make it
      
      *** CMake build failed
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for lightgbm
Failed to build lightgbm
ERROR: Could not build wheels for lightgbm, which is required to install pyproject.toml-based projects`

**Comment by jarichb** [comment]
Yes, sounds good to me! Thank you
